### ISSUE TITLE
Error with ESP8266 Arduino core >2.5: "ISR not in IRAM!"

### ISSUE THREAD

**Issue by marcelstoer** [PaulStoffregen/XPT2046_Touchscreen#25]
### Description

Reference: https://www.esp8266.com/viewtopic.php?f=32&t=19797

https://github.com/esp8266/Arduino/pull/5995 introduced a new check for ISRs. This library now fails at `XPT2046_Touchscreen::begin()` (and possibly other functions).

Side note: for an example how others deal with this see https://github.com/arendst/Sonoff-Tasmota/commit/e5f8167ebe70d658d220e78168f6069ac740e84f

### Steps To Reproduce Problem

Run the sketch below.

### Hardware & Software

Any ESP8266 with Arduino core 2.5.1 or up; I used a WeMOS D1 mini Pro.

### Arduino Sketch

```cpp
#include <Arduino.h>
#include <SPI.h>
#include <XPT2046_Touchscreen.h>
#include <ILI9341_SPI.h>

#define TFT_DC D2
#define TFT_CS D1
#define TFT_LED D8

#define TOUCH_CS D3
#define TOUCH_IRQ  D4

ILI9341_SPI tft = ILI9341_SPI(TFT_CS, TFT_DC);
XPT2046_Touchscreen ts(TOUCH_CS, TOUCH_IRQ);
TouchControllerWS touchController(&ts);

void setup() {
  Serial.begin(115200);

  // The LED pin needs to set HIGH
  // Use this pin to save energy
  // Turn on the background LED
  Serial.println(TFT_LED);
  pinMode(TFT_LED, OUTPUT);
  digitalWrite(TFT_LED, HIGH);    // HIGH to Turn on;

  Serial.println("Initializing touch screen...");
  ts.begin(); // this is XPT2046_Touchscreen::begin() in the stacktrace
}

void loop() {
}
```

### Errors or Incorrect Output

It will fail with "ISR not in IRAM!". Decoded stacktrace:

```
0x4020eb15: Print::write(char const*) at /Users/marcelstoer/Library/Arduino15/packages/esp8266/hardware/esp8266/2.5.2/cores/esp8266/Print.h line 60
0x401004be: __attachInterrupt(uint8_t, voidFuncPtr, int) at /Users/marcelstoer/Library/Arduino15/packages/esp8266/hardware/esp8266/2.5.2/cores/esp8266/core_esp8266_wiring_digital.cpp line 206
0x402053c0: XPT2046_Touchscreen::begin() at /Users/marcelstoer/Data/Arduino/libraries/XPT2046_Touchscreen/XPT2046_Touchscreen.cpp line 41
```

**Comment by marcelstoer** [comment]
Another example of a valid fix: https://github.com/sandeepmistry/arduino-LoRa/pull/257

**Comment by marcelstoer** [comment]
Argh, I failed to notice there's already a PR that attempts to fix this: #24.
